executor/common_kvm_arm64.h

```diff
@@ -195,6 +196,14 @@ static void setup_cpu_with_opts(int vmfd, int cpufd, const struct kvm_opt* opt,
 	init.features[0] = features;
 	// Use the modified struct kvm_vcpu_init to initialize the virtual CPU.
 	ioctl(cpufd, KVM_ARM_VCPU_INIT, &init);
+        // Make sure that PMUs are initialized if enabled
+        if (opt_count == 1 && (features & pmu_bit)) {
+                struct kvm_device_attr attrs = {
+                    .group = KVM_ARM_VCPU_PMU_V3_CTRL,
+                    .attr = KVM_ARM_VCPU_PMU_V3_INIT
```

Comment:
Hmm, according to https://www.kernel.org/doc/Documentation/virt/kvm/devices/vcpu.rst:

> If using the PMUv3 with an in-kernel virtual GIC implementation, this must be done after initializing the in-kernel irqchip.

So looks like this call has to follow `syz_kvm_vgic_v3_setup()`.

Do you have a syzlang program working for you with this change?
Maybe we can just factor `ioctl$KVM_SET_DEVICE_ATTR()` outside of `syz_kvm_add_vcpu()` in that program?

Reply:
Agreed, decoupling it from `syz_kvm_add_vcpu()` sounds good (I think it'll be quite clear using a flag as suggested above).
A sample program that works with this change is:
```
r0 = openat$kvm(0, &AUTO='/dev/kvm\x00', 0x0, 0x0)
r1 = ioctl$KVM_CREATE_VM(r0, AUTO, 0x0)
r2 = syz_kvm_setup_syzos_vm(r1, &(0x7f0000c00000/0x400000)=nil)
r3 = syz_kvm_add_vcpu(r2, &AUTO={0x0, &AUTO=[@msr={AUTO, AUTO, {0x603000000013c4f1, 0x8000}}], AUTO}, &AUTO=[@featur1={0x1, 0x8}], 0x1)
ioctl$KVM_RUN(r3, 0xae80, 0x0)
```